code-challenges.test.js

```diff
@@ -13,44 +15,132 @@
-// Expected output: [-9, 7, 9, 199]
+describe("leasttoGreatest", () => {
+    it("takes in an array and returns a new array of only odd numbers sorted from least to greatest", () => {
+        const fullArr1 = [4, 9, 0, "7", 8, true, "hey", 7, 199, -9, false, "hola"]
+         // expected output = [-9, 7, 9, 199]
+        
+        const fullArr2 = ["hello", 7, 23, -823, false, 78, null, "67", 6, "number"]
+        // expected output: [-823, 7, 23]
+    expect(leasttoGreatest(fullArr1)).toEqual([-9, 7, 9, 199])
+    expect(leasttoGreatest(fullArr2)).toEqual([-823, 7, 23])
+    })
+  })
 
-const fullArr2 = ["hello", 7, 23, -823, false, 78, null, "67", 6, "number"]
-// Expected output: [-823, 7, 23]
+  //    ReferenceError: leasttoGreatest is not defined
 
 
 // b) Create the function that makes the test pass.
+  
+  //Pseudo code:
+    // declare a function called leasttoGreatest
+    // use .filter to get a new array 
+    //  use .sort with two arguments (a,b) to sort the new from least to greatest
+
+
+const leasttoGreatest = (array) => {
+  const newArray = array.filter (value => typeof
+    value === "number" && value % 2 !== 0
+  ); newArray.sort((a,b) => {return a - b})
+    return newArray
+} 
+
+// console.log (leasttoGreatest(fullArr1))
+
+// Test Suites: 1 passed, 1 total
+
+
 
 
 // --------------------3) Create a function that takes in an array and returns an array of the accumulating sum. An empty array should return an empty array.
 
 // a) Create a test with expect statements for each of the variables provided.
 
-const numbersToAdd1 = [2, 4, 45, 9]
-// Excpected output: [2, 6, 51, 60]
+describe("accumulateSum", () => {
+    it("takes in an array and returns an array of the accumulating sum. An empty array should return an empty array", () => {
+        const numbersToAdd1 = [2, 4, 45, 9]
+        // Excpected output: [2, 6, 51, 60]
+
+        const numbersToAdd2 = [0, 7, -8, 12]
+        // Expected output: [0, 7, -1, 11]
 
-const numbersToAdd2 = [0, 7, -8, 12]
-// Expected output: [0, 7, -1, 11]
+        const numbersToAdd3 = []
+        // Expected output: []
+    expect(accumulateSum(numbersToAdd1)).toEqual([2, 6, 51, 60])
+    expect(accumulateSum(numbersToAdd2)).toEqual([0, 7, -1, 11])
+    expect(accumulateSum(numbersToAdd3)).toEqual([])
+    })
+  })
 
-const numbersToAdd3 = []
-// Expected output: []
+//   ReferenceError: accumulateSum is not defined
 
 
 // b) Create the function that makes the test pass.
+
+  //Psuedo code:
+    // declare a function called accumulateSum 
+    // declare a variable called sum and assign it to 0 
+    // use .map to return an array with the same length 
+    // use addition assignment (+=) to add the values with each iteration
+
+
+
+    const accumulateSum = (array) => {
+        let sum = 0;
+       return array.map (value => {
+            return  sum += value
+        })
+
+    }
+// console.log (accumulateSum(numbersToAdd1))
+
+// Test Suites: 1 passed, 1 total
```

Comment:
Will, again, you are killing these coding challenges! This is good stuff.  Great, clean, efficient code.  You even used .map on #3 (and didn't speak up in class. :). Show off that work! ) Seriously, you are getting better and better.  Your hard work is showing.  One thing, don't let up on your pseudocode.  Write out as much as you can, it will only benefit you as you encounter more and more challenging problems.  Super impressed!